src/scripts/Providers/OSUI/Datepicker/Flatpickr/AbstractFlatpickr.ts

```diff
@@ -249,6 +266,28 @@ namespace Providers.OSUI.Datepicker.Flatpickr {
 			}
 		}
 
+		/**
+		 * Method used to add the custom callback to the OnOpen event and add the tabindex for the today button
+		 *
+		 * @protected
+		 * @memberof AbstractFlatpickr
+		 */
+		protected onDatePickerOpen(): void {
+			// Add the tabindex for the link inside the today button if it exists
+			if (this.configs.ShowTodayButton && this._todayButtonElem) {
+				OSFramework.OSUI.Helper.A11Y.TabIndexTrue(
+					this._todayButtonElem.querySelector(OSFramework.OSUI.GlobalEnum.HTMLElement.Link)
+				);
+			}
+			// Call the custom callback provided for users to add custom code when a datepicker opens
+			if (
+				this.onOpenCustomCallback !== undefined &&
+				typeof this.onOpenCustomCallback === OSFramework.OSUI.Constants.JavaScriptTypes.Function
+			) {
+				this.onOpenCustomCallback(this.provider);
```

Comment:
This a new feature, correct?
If so, should it be implemented in the context of this task?
Also, shouldn't it be asynchronously invoked?

Reply:
@rugoncalves Adding this new event was the only way I found to change the `tabindex` while opening the date pickers (after the first time) and followed the same approach used by OnClose to call the provider's event.
About being asynchronous why would we need that?

Reply:
@BenOsodrac any context on this since I followed the same approach as for `OnClose`? 
I assume that it was implemented thinking only about an internal usage of the events and so it's synchronous, but have no context.

Reply:
I think its ok yes, close and open are just internal events on JS, so no need to be async.